src/fd.zig

```diff
@@ -247,15 +247,6 @@ pub const FDImpl = packed struct {
                     else => null,
                 };
             },
-            .mac => result: {
-                const fd = this.encode();
-                bun.assert(fd != bun.invalid_fd);
-                bun.assert(fd.cast() >= 0);
-                break :result switch (bun.C.getErrno(bun.sys.syscall.@"close$NOCANCEL"(fd.cast()))) {
```

Comment:
These two calls are not identical. `close$NOCANCEL` is deliberate

Reply:
`std.c.close` on macos was changed to be `close$NOCANCEL`